lib/store/ca_store.go

```diff
@@ -448,6 +448,107 @@ func (s *CAStore) writeDrainItemToDisk(entry *cache.MemoryEntry) error {
 	return nil
 }
 
+// GetCacheFileReader overrides cacheStore.GetCacheFileReader to check
+// memory cache first before reading from disk.
+func (s *CAStore) GetCacheFileReader(name string) (FileReader, error) {
+	// Check memory cache first
+	if s.memCache != nil {
+		if entry := s.memCache.Get(name); entry != nil {
+			return NewBufferFileReader(entry.Data), nil
+		}
+	}
+
+	return s.cacheStore.GetCacheFileReader(name)
+}
+
+// GetCacheFileMetadata overrides cacheStore.GetCacheFileMetadata to serve
+// TorrentMeta from memory cache when available.
+func (s *CAStore) GetCacheFileMetadata(name string, md metadata.Metadata) error {
+	// Only serve TorrentMeta from memory cache
+	if s.memCache != nil && md.GetSuffix() == "_torrentmeta" {
+		if entry := s.memCache.Get(name); entry != nil {
+			if entry.MetaInfo == nil {
+				// Entry exists but has no MetaInfo
+				return fmt.Errorf("entry %s doesn't have any metainfo", entry.Name)
+			}
+			// Serialize and deserialize for consistency with disk behavior
+			b, err := entry.MetaInfo.Serialize()
+			if err != nil {
+				return fmt.Errorf("serialize metainfo: %s", err)
+			}
+			return md.Deserialize(b)
+		}
+	}
+
+	// Fallback to disk
+	return s.cacheStore.GetCacheFileMetadata(name, md)
+}
+
+// GetCacheFileStat overrides cacheStore.GetCacheFileStat to check memory cache first.
+func (s *CAStore) GetCacheFileStat(name string) (os.FileInfo, error) {
+	// Check memory cache first
+	if s.memCache != nil {
+		if entry := s.memCache.Get(name); entry != nil {
+			return &memoryFileInfo{
+				name:    name,
+				size:    int64(entry.Size()),
+				modTime: entry.CreatedAt,
+			}, nil
+		}
+	}
+
+	// Fallback to disk
+	return s.cacheStore.GetCacheFileStat(name)
+}
+
+// ListCacheFiles overrides cacheStore.ListCacheFiles to include memory cache entries.
+func (s *CAStore) ListCacheFiles() ([]string, error) {
+	// Get files from disk
+	diskFiles, err := s.cacheStore.ListCacheFiles()
+	if err != nil {
+		return nil, err
+	}
+
+	// If no memory cache, return disk files only
+	if s.memCache == nil {
+		return diskFiles, nil
+	}
+
+	// Merge memory cache entries with disk files
+	memNames := s.memCache.ListNames()
+
+	// Use map to deduplicate (blob might be in both memory and disk)
+	allFiles := make(map[string]bool)
+	for _, name := range diskFiles {
+		allFiles[name] = true
+	}
+	for _, name := range memNames {
+		allFiles[name] = true
+	}
+
+	// Convert back to slice
+	result := make([]string, 0, len(allFiles))
+	for name := range allFiles {
+		result = append(result, name)
+	}
+
+	return result, nil
+}
+
+// memoryFileInfo implements os.FileInfo for memory cache entries.
```

Comment:
nit: replace comment with a compile time check, which also documents the contract that the struct must satisfy the interface:
```suggestion
var _ os.FileInfo = &memoryFileInfo{}
```